find_email_address2.py

```diff
@@ -0,0 +1,62 @@
+import sys, re, urllib
+from bs4 import BeautifulSoup
+from contextlib import closing
+from selenium.webdriver import Firefox
+from selenium.webdriver.support.ui import WebDriverWait
+
+visitedDomains = set()
+
+def find_email_address2(domain):
+    #if the print stays here it prints before the check
+    if domain in visitedDomains:
+        return
+    #print("visiting: "+domain)
+    with closing(Firefox()) as browser:
+        browser.get('http://'+domain)
+        page_source = browser.page_source
+    soup = BeautifulSoup(page_source, "lxml")
+    #find email addresses on page
+    domains = check_page(soup)
+    visitedDomains.add(domain)
+    for i in set(domains):
+        print(i)
+    #find links in page
+    links = find_links(soup, domain)
+    #print(links)
+    for j in set(links):
+        find_email_address2(j)
+
+def check_page(document):
+    email_reg = r"[a-z0-9\.\-+_]+@[a-z0-9\.\-+_]+\.[a-z]+"
+    pattern = re.compile(email_reg)
+    return re.findall(pattern, str(document))
+
+def find_links(document, domain):
+    links = re.findall('''href=["'](.[^"']+)["']''', str(document))
```

Comment:
There are times (particularly on jana.com) where a link doesn't look like this, since it's a single page app, and you're just changing the route within that.  How would you check for that?  I think this is why you're not ever navigating to jana.com/careers